builder/dockerfile/parser/parser.go

```diff
@@ -101,19 +103,24 @@ func parseLine(line string) (string, *Node, error) {
 // Parse is the main parse routine.
 // It handles an io.ReadWriteCloser and returns the root of the AST.
 func Parse(rwc io.Reader) (*Node, error) {
+	currentLine := 0
 	root := &Node{}
+	root.StartLine = currentLine
 	scanner := bufio.NewScanner(rwc)
 
 	for scanner.Scan() {
 		scannedLine := strings.TrimLeftFunc(scanner.Text(), unicode.IsSpace)
+		currentLine++
 		line, child, err := parseLine(scannedLine)
 		if err != nil {
 			return nil, err
```

Comment:
Now I'm setting the Endline always as the same as the last child's Endline. Even if it returns early, it should make sense to set the Endline as the last parsed line.